scripts/sync_tensorflow_protobuf_stubs.sh

```diff
@@ -1,73 +1,102 @@
 
-pip install pre-commit mypy-protobuf=="$MYPY_PROTOBUF_VERSION"
+if uname -a | grep Darwin; then
+    # brew install coreutils wget
+    PLAT=osx
+else
+    PLAT=linux
+fi
+REPO_ROOT="$(realpath "$(dirname "${BASH_SOURCE[0]}")"/..)"
+TMP_DIR="$(mktemp -d)"
+TENSORFLOW_FILENAME="v$TENSORFLOW_VERSION.zip"
+PROTOC_FILENAME="protoc-$PROTOBUF_VERSION-$PLAT-x86_64.zip"
+PROTOC_URL="https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOBUF_VERSION/$PROTOC_FILENAME"
+TENSORFLOW_URL="https://github.com/tensorflow/tensorflow/archive/refs/tags/$TENSORFLOW_FILENAME"
 
-cd "$(dirname "$0")" > /dev/null
-cd ../stubs/tensorflow
-mkdir -p repository
-pushd repository &> /dev/null
-    # If the script fails halfway, it's nice to be able to re-run it immediately
-    if [ ! -d "tensorflow" ] ; then
-        git clone --depth 1 --branch v"$TENSORFLOW_VERSION" https://github.com/tensorflow/tensorflow.git
-    fi
-    pushd tensorflow &> /dev/null
-        # Folders here cover the more commonly used protobufs externally and
-        # their dependencies. Tensorflow has more protobufs and can be added if requested.
-        protoc --mypy_out "relax_strict_optional_primitives:$REPO_ROOT/stubs/tensorflow" \
-            tensorflow/compiler/xla/*.proto \
-            tensorflow/compiler/xla/service/*.proto \
-            tensorflow/core/example/*.proto \
-            tensorflow/core/framework/*.proto \
-            tensorflow/core/protobuf/*.proto \
-            tensorflow/core/protobuf/tpu/*.proto \
-            tensorflow/core/util/*.proto \
-            tensorflow/python/keras/protobuf/*.proto \
-            tensorflow/tsl/protobuf/*.proto
-    popd &> /dev/null
-popd &> /dev/null
+cd "$TMP_DIR"
+echo "Working in $TMP_DIR"
 
-# These protos exist in a folder with protos used in python, but are not
-# included in the python wheel. They are likely only used for other
-# language builds. stubtest was used to identify them by looking for
-# ModuleNotFoundError.
-rm tensorflow/compiler/xla/service/hlo_execution_profile_data_pb2.pyi \
-   tensorflow/compiler/xla/service/hlo_profile_printer_data_pb2.pyi \
-   tensorflow/compiler/xla/service/test_compilation_environment_pb2.pyi \
-   tensorflow/compiler/xla/xla_pb2.pyi \
-   tensorflow/core/protobuf/autotuning_pb2.pyi \
-   tensorflow/core/protobuf/conv_autotuning_pb2.pyi \
-   tensorflow/core/protobuf/critical_section_pb2.pyi \
-   tensorflow/core/protobuf/eager_service_pb2.pyi \
-   tensorflow/core/protobuf/master_pb2.pyi \
-   tensorflow/core/protobuf/master_service_pb2.pyi \
-   tensorflow/core/protobuf/replay_log_pb2.pyi \
-   tensorflow/core/protobuf/tpu/compile_metadata_pb2.pyi \
-   tensorflow/core/protobuf/worker_pb2.pyi \
-   tensorflow/core/protobuf/worker_service_pb2.pyi \
-   tensorflow/core/util/example_proto_fast_parsing_test_pb2.pyi
+# Install protoc
+wget "$PROTOC_URL"
+mkdir protoc_install
+unzip "$PROTOC_FILENAME" -d protoc_install
+protoc_install/bin/protoc --version
```

Comment:
Might it make sense to install protoc in the venv so that a call to `protoc` directly picks up the version we just downloaded instead of the system `protoc`?